### PR TITLE
Benchmark backprop of a te.TransformerLayer.

### PR DESCRIPTION
**Pull request by wujingyue**
```
$ mpirun -np 1 pytest tests/python/test_transformer_engine.py --only-mpi

------------------------------------------------------------------------------------------------- benchmark: 2 tests ------------------------------------------------------------------------------------------------
Name (time in us)                           Min                   Max                  Mean              StdDev                Median                 IQR            Outliers       OPS            Rounds  Iterations
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
test_transformer_layer[forward]        952.7383 (1.0)      1,200.1079 (1.0)      1,016.5755 (1.0)      103.4595 (1.0)        976.7041 (1.0)       79.6057 (1.0)           1;1  983.6948 (1.0)           5           1
test_transformer_layer[backward]     1,070.4808 (1.12)     1,347.0454 (1.12)     1,188.3112 (1.17)     136.6049 (1.32)     1,125.9178 (1.15)     257.2080 (3.23)          1;0  841.5304 (0.86)          5           1
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

### CODE REVIEW

**Comment by wujingyue** [comment]
!build

File: tests/python/test_transformer_engine.py [hunk 1]
```diff
@@ -53,7 +64,32 @@ def test_transformer_layer(multidevice_test):
...
-    y = transformer_layer(x)
-    assert y.size() == torch.Size([batch_size, sequence_length, hidden_size])
+
+    match compute_type:
+        case ComputeType.FORWARD:
+
+            def benchmark_fn():
+                return transformer_layer(x)
+
+            y = benchmark(benchmark_fn)
```
[the remark this answers is not in the document]
Comment (wujingyue) [reply tests/python/test_transformer_engine.py:74]: Yes. I added an example in the PR description.

**Comment by wujingyue** [comment]
!build

File: tests/python/test_transformer_engine.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (cowanmeg) [reply tests/python/test_transformer_engine.py:72]: Awesome! Thank you!

**Review by cowanmeg — APPROVED** [review]

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
I'm getting [a weird error](https://github.com/NVIDIA/Fuser/actions/runs/10932565435/job/30349609386) from jit_python_distributed_tests_17_A100 based on this PR. I'll fix that before merging this PR. I've yet to reproduce that locally unfortunately.

**Comment by wujingyue** [comment]
I believe the error has something to do with calling init_process_group and destroy_process_group for each test. Due to race conditions, some rank calls init_process_group from the second test before all ranks finished destroying the default process group.

**Comment by cowanmeg** [comment]
How about we don't add this to CI for now? We can benchmark locally and then work on fixing the CI problems

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
!build

**Pull request merged**